packages/graphqlsp/src/index.ts

```diff
@@ -179,7 +181,106 @@ function create(info: ts.server.PluginCreateInfo) {
       return result;
     });
 
-    if (!newDiagnostics.length) {
+    const imports = findAllImports(source);
+    if (imports.length && shouldCheckForColocatedFragments) {
+      const typeChecker = info.languageService.getProgram()?.getTypeChecker();
+      imports.forEach(imp => {
+        if (!imp.importClause) return;
+
+        const importedNames: string[] = [];
+        if (imp.importClause.name) {
+          importedNames.push(imp.importClause?.name.text);
+        }
+
+        if (
+          imp.importClause.namedBindings &&
+          isNamespaceImport(imp.importClause.namedBindings)
+        ) {
+          // TODO: we might need to warn here when the fragment is unused as a namespace import
+          return;
+        } else if (
+          imp.importClause.namedBindings &&
+          isNamedImportBindings(imp.importClause.namedBindings)
+        ) {
+          imp.importClause.namedBindings.elements.forEach(el => {
+            importedNames.push(el.name.text);
+          });
+        }
+
+        const symbol = typeChecker?.getSymbolAtLocation(imp.moduleSpecifier);
+        if (!symbol) return;
+
+        const moduleExports = typeChecker?.getExportsOfModule(symbol);
+        if (!moduleExports) return;
+
+        const missingImports = moduleExports
+          .map(exp => {
+            if (importedNames.includes(exp.name)) {
+              return;
+            }
+
+            const declarations = exp.getDeclarations();
+            const declaration = declarations?.find(x => {
+              // TODO: check whether the sourceFile.fileName resembles the module
+              // specifier
+              return true;
+            });
+
+            if (!declaration) return;
+
+            const [template] = findAllTaggedTemplateNodes(declaration);
+            if (template) {
+              let node = template;
+              if (
+                isNoSubstitutionTemplateLiteral(node) ||
+                isTemplateExpression(node)
+              ) {
+                if (isTaggedTemplateExpression(node.parent)) {
+                  node = node.parent;
+                } else {
+                  return;
+                }
+              }
+
+              const text = resolveTemplate(
+                node,
+                node.getSourceFile().fileName,
+                info
+              );
+              const parsed = parse(text);
+              if (
+                parsed.definitions.every(
+                  x => x.kind === Kind.FRAGMENT_DEFINITION
+                )
+              ) {
+                return `'${exp.name}'`;
+              }
+            }
+          })
+          .filter(Boolean);
+
+        if (missingImports.length) {
+          newDiagnostics.push({
+            file: source,
+            length: imp.getText().length,
+            start: imp.getStart(),
+            category: ts.DiagnosticCategory.Message,
+            code: 51001,
```

Comment:
I have to check whether this is related to VSCode or the `code` here but category message is visualised just like an error 😅